internal/controller/dnsrecord_controller.go

```diff
@@ -264,6 +263,22 @@ func generationChanged(record *v1alpha1.DNSRecord) bool {
 	return record.Generation != record.Status.ObservedGeneration
 }
 
+// exponentialRequeueTime consumes the current time and doubles it until it reaches defaultRequeueTime
+func exponentialRequeueTime(lastRequeueTime string) time.Duration {
+	lastRequeue, err := time.ParseDuration(lastRequeueTime)
+	// corrupted DNSRecord. This value naturally set only via time.Duration.String() call
+	if err != nil {
+		// default to the least confidence timeout
+		return validationRequeueTime
+	}
+	// double the duration. Return the max timeout if overshoot
+	newReqeueue := lastRequeue * 2
```

Comment:
we don't want to just take the last requeue time and double it, as the initial value is randomized. So we need to work off of the initial values, e.g.

requeueTime := min(max_requeue, math.Pow(requeueDuration, math.Pow(2, requeueCount)))

Reply:
I'd have it double instead of raising to the power - makes it grow too fast IMO. 
The double sequence will be (seconds): 5, 10, 20, 40, 80, 160, 320, 640, 900
While the power 2 sequence: is 5, 25, 625, 900. 
And we don't keep the `requeueCount`. I'll reset the timeout on the successful validation to ensure we are not doubling the randomized value.

Reply:
ah, I see, if we aren't keeping the requeueCount then yes, we cannot do it this way.